### ISSUE TITLE
Minilmv2 implement detail of multi-head self-attention relation distillation?

### ISSUE THREAD

**Issue by trouble-maker007** [microsoft/unilm#349]
**Describe**
Model I am using (Minilmv2):
hello, @WenhuiWang0824 @donglixp, I want to reproduce minilmv2 in other language, I have some questions about the multi-head self-attention relation distillation:
1. If the teacher and student model with the different attention heads, following the paper: **. In
order to obtain self-attention vectors (queries, keys and values) of multiple relation heads, we first concatenate
self-attention vectors of different attention heads and then split them according to the desired number of relation
heads.**, 

Scalar dimensions referenced here:
  | #   B = batch size (number of sequences)
  | #   F = sequence length
  | #   N = `num_attention_heads`
  | #   H = `size_per_head`
  | #   R = `num_relation_heads`

`query_layer` = [B, N, F, H], is it mean that suppose the teacher(large model) `query_layer` = [16, 16, 64, 64] and the student `query_layer` = [16, 12, 64, 64], and then change the teacher and student head to 64(num_relation_heads), just like [16, 64, 64, x], x is refer to different size of `size_per_head`? and then each carries its own proceed to q-q self attention relation to get the relation score , then transfer by KL-Divergence loss ? Following is my sample code for `query` and `query`:

```
# `query_layer` = [B, N, F, H], B = 16, N = 16, F = 64, H = 64, R = 64
teacher_query = torch.rand(16, 16, 64, 64)
teacher_query = torch.reshape(teacher_query, (16, -1))
teacher_query = torch.reshape(teacher_query, (16, 64, 64, -1))
teacher_qq_relation_scores = torch.matmul(teacher_query, teacher_query.transpose(-1, -2))
teacher_qq_relation_scores = teacher_qq_relation_scores / math.sqrt(16) # now teacher relation head size is 16
teacher_qq_relation_scores = teacher_qq_relation_scores + attention_mask # whether need to add the attention_mask?
teacher_qq_relation_probs = nn.Softmax(dim=-1)(teacher_qq_relation_scores)
print(teacher_qq_relation_probs) ->  torch.Size([16, 64, 64, 64])

student_query = torch.rand(16, 12, 64, 64)
student_query = torch.reshape(student_query, (16, -1))
student_query = torch.reshape(teacher_query, (16, 64, 64, -1))
student_qq_relation_scores = torch.matmul(teacher_query, teacher_query.transpose(-1, -2))
student_qq_relation_scores = student_qq_relation_scores / math.sqrt(12) # now student relation head size is 12
student_qq_relation_scores = student_qq_relation_scores + attention_mask # whether need to add the attention_mask?
student_qq_relation_probs = nn.Softmax(dim=-1)(student_qq_relation_scores)
print(student_qq_relation_probs) ->  torch.Size([16, 64, 64, 64])

kd_loss = nn.KLDivLoss()
qq_transfer_loss = kd_loss(student_qq_relation_probs, teacher_qq_relation_probs)
```

2.the `student_qq_relation_scores = student_qq_relation_scores + attention_mask` need to add `attention_mask` or not?

2. the attenttion probs no need to go by dropout `teacher_qq_relation_probs = self.dropout(teacher_qq_relation_probs)`?Suppose just with teacher query and query self attention relation, Is it use `teacher_qq_relation_probs = nn.Softmax(dim=-1)(teacher_qq_relation_scores)` to transfer by KL-Divergence loss ?
